src/vsr/replica.zig

```diff
@@ -3092,6 +2932,29 @@ pub fn ReplicaType(
             }
             self.view_headers.verify();
 
+            const BitSet = std.bit_set.IntegerBitSet(128);
+            assert(BitSet.MaskInt == std.meta.fieldInfo(Header, .context).field_type);
+
+            var missing = BitSet.initEmpty();
+            if (command == .do_view_change) {
+                for (self.view_headers.array.constSlice()) |*header, i| {
+                    const slot = self.journal.slot_for_op(header.op);
+                    if (self.journal.header_for_op(header.op)) |h| {
+                        if (h.checksum == header.checksum) {
+                            if (self.journal.dirty.bit(slot) and !self.journal.faulty.bit(slot)) {
+                                missing.set(i);
+                            }
+                        } else {
+                            missing.set(i);
+                        }
+                    } else {
+                        if (!self.journal.faulty.bit(slot)) {
+                            missing.set(i);
+                        }
+                    }
```

Comment:
Also, the difference between `header_for_op/header_with_op` is subtle. I woneder if we should use `header_for_slot` instead?

Reply:
Also, for case2, can we assert that that's an older prepare? That it is either for a different wrap, or has a smaller view? (not sure if both are possible here)

Reply:
> Also, for case2, can we assert that that's an older prepare? That it is either for a different wrap, or has a smaller view? (not sure if both are possible here)

Tricky case. I think we can't assert this, since successive view-changes can swing the version of an op forward-and-back, and crash/recovery restores the journal state. So even though the header in the view_headers is logically newer — i.e. part of a higher log_view — it might have originated in an older view.